stage1/rootfs/prepare-app/prepare-app.c

```diff
@@ -179,5 +179,15 @@ int main(int argc, char *argv[])
 	pexit_if(symlink("/dev/pts/ptmx", to) == -1,
 		"Failed to create /dev/ptmx symlink");
 
+	/* diagexec */
+	exit_if(snprintf(to, sizeof(to), "%s/.diagexec", root) >= sizeof(to),
+		"Path too long: \"%s\"", to);
+	fd = open(to, O_WRONLY|O_CREAT|O_CLOEXEC|O_NOCTTY, 0755);
+	if (fd != -1)
+		close(fd);
+	pexit_if(mount("/diagexec", to, "bind", MS_BIND, NULL) == -1,
```

Comment:
I am not really keen on this; are we so sure that we will need this ability to impose this arbitrary constraint in the spec? I guess if we had example generic use cases that might help convince me.